Search/ReIndex/ResumeManager.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/*
+ * This file is part of the MassiveSearchBundle
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search\ReIndex;
+
+/**
+ * Allow the storage of "checkpoints" which indicate how far
+ * a reindexing process has progressed in the case that it is
+ * interupted (f.e. by an error) with the aim of allowing the
+ * reindexing process to resume from where it left off.
+ *
+ * This is basically a key/value store.
+ */
+class ResumeManager
+{
+    /**
+     * Store a checkpoint. This would typically be an offset
+     * from where the reindexer can subsequently resume its
+     * task.
+     *
+     * @param mixed $name
+     * @param mixed $value
+     */
+    public function setCheckpoint($name, $value)
```

Comment:
But that's not a good thing... I.e. we have projects where loads of assets are uploaded, so this functionality would be even more important for the assets in these projects. Would it be hard to implement the resume functionality directly in here? Would also make more sense IMO.